File: GoL/Model/Figure.m
//
//  Figure.m
//  GoL
//
//  Created by Georg Meyer on 01.06.14.
//
//

#import "Figure.h"

@implementation Figure

+ (Figure *)createCross {
    Figure *cross = [[self alloc] init];

    cross.dimensions = [[Dimensions alloc] initWithHeight:3 andWidth:3];
    cross.field = [[NSMutableArray alloc] initWithCapacity:cross.dimensions.height];

    [cross addLine: @"_o_"];
    [cross addLine: @"_o_"];
    [cross addLine: @"_o_"];

    return cross;
}

+ (Figure *)createLightweightSpaceship {
    Figure *cross = [[self alloc] init];

    cross.dimensions = [[Dimensions alloc] initWithHeight:4 andWidth:5];
    cross.field = [[NSMutableArray alloc] initWithCapacity:cross.dimensions.height];

    [cross addLine: @"o__o_"];
    [cross addLine: @"____o"];
    [cross addLine: @"o___o"];
    [cross addLine: @"_oooo"];
    return cross;
}


- (void)addLine:(NSString *)line {
    NSMutableArray *row = [[NSMutableArray alloc] initWithCapacity:self.dimensions.width];

    for(int i = 0; i < line.length; i++) {
        NSRange range = NSMakeRange(i, 1);

        [row addObject :[NSNumber numberWithBool:[[line substringWithRange:range] isEqualToString:@"o"]]];
    }

    [self.field addObject:row];
}

@end

File: GoL/Model/Figure.swift
//
//  Figure.swift
//  GoL
//
//  Created by Georg Meyer on 03.06.14.
//
//

import Foundation

extension Model {
    class Figure {
        var dimensions : Dimensions
        var field : Bool[][]
        
        init(dimensions : Dimensions){
            self.field = []
            self.dimensions = dimensions
        }
        
        func addLine(line : String) {
            var row : Bool[] = []
            
            let o : Character = "o"
            for char in line {
                row.append(o == char)
            }
            
            field.append(row)
        }
        
        class func createLightweightSpaceship() -> Figure {
            let dimensions = Dimensions(width: 5, height: 4)
            let figure = Figure(dimensions: dimensions)
            
            figure.addLine("o__o_")
            figure.addLine("____o")
            figure.addLine("o___o")
            figure.addLine("_oooo")
            
            return figure
        }
    }
}